addons/sourcemod/scripting/gokz-replays/recording.sp

```diff
@@ -119,45 +149,84 @@ void OnPlayerRunCmdPost_Recording(int client, int buttons, int tickCount, const
 	recordedRecentData[client].SetArray(tick, tickData);
 }
 
-void GOKZ_OnTimerStart_Recording(int client)
+Action GOKZ_OnTimerStart_Recording(int client)
+{
+	// Hack to fix an exception when starting the timer on the very
+	// first tick after loading the plugin.
+	if (recordedRecentData[client].Length == 0)
+	{
+		return Plugin_Handled;
+	}
+
+	return Plugin_Continue;
+}
+
+void GOKZ_OnTimerStart_Post_Recording(int client)
 {
 	timerRunning[client] = true;
 	StartRunRecording(client);
 }
 
 void GOKZ_OnTimerEnd_Recording(int client, int course, float time, int teleportsUsed)
 {
-	DataPack dp = new DataPack();
-	dp.WriteCell(client);
-	dp.WriteCell(course);
-	dp.WriteFloat(time);
-	dp.WriteCell(teleportsUsed);
-	delete recordedPostRunData[client];
-	recordedPostRunData[client] = recordedRunData[client];
-	recordedRunData[client] = new ArrayList(sizeof(ReplayTickData));
+	DataPack data = new DataPack();
+	data.WriteCell(GetClientUserId(client));
+	data.WriteCell(course);
+	data.WriteFloat(time);
+	data.WriteCell(teleportsUsed);
+
 	if (timerRunning[client])
 	{
-   		CreateTimer(RP_PLAYBACK_BREATHER_TIME, EndRecording, dp);
-   		postRunRecording[client] = true;
-   		timerRunning[client] = false;
+		// The previous run breather still did not finish, end it now or
+		// we will start overwriting the data.
+		if (runningRunBreatherTimer[client] != INVALID_HANDLE)
+		{
+			TriggerTimer(runningRunBreatherTimer[client], false);
+		}
+
+		timerRunning[client] = false;
+		postRunRecording[client] = true;
+
+		// Swap recordedRunData and recordedPostRunData.
+		// This lets new runs start immediately, before the post-run breather is
+		// finished recording.
+		ArrayList tmp = recordedPostRunData[client];
+		recordedPostRunData[client] = recordedRunData[client];
+		recordedRunData[client] = tmp;
+		recordedRunData[client].Clear();
```

Comment:
Any particulat reason that we swap them instead of deleting/re-creating? Do you think it's faster?